lib/nostrum/struct/guild/audit_log.ex

```diff
@@ -0,0 +1,38 @@
+defmodule Nostrum.Struct.Guild.AuditLog do
+  @moduledoc """
+  Represents a guild's audit log.
+  """
+
+  alias Nostrum.Struct.Guild.AuditLogEntry
+  alias Nostrum.Struct.{User, Webhook}
+  alias Nostrum.Util
+
+  defstruct [:entries, :users, :webhooks]
+
+  @typedoc "Entries of this guild's audit log."
+  @type entries :: [AuditLogEntry.t()]
+
+  @typedoc "Users found in the audit log."
+  @type users :: [User.t()]
+
+  @typedoc "Webhooks found in the audit log."
+  @type webhooks :: [Webhook.t()]
+
+  @type t :: %__MODULE__{
+          entries: entries,
+          users: users,
+          webhooks: webhooks
+        }
+
+  @doc false
+  def to_struct(map) do
+    struct(
+      __MODULE__,
```

Comment:
Any particular reason you're doing the updates this way? Discord isn't exactly known to be super faithful about the fields they send. I'd suggest you check out how any other struct modules does this and see if it'd be better to do it similarly here, either just to be more consistent, or to handle any possible missing keys.

Reply:
This seemed the most intuitive - I see the point about missing keys though, maybe I should handle that more gracefully. Also, I could just use `%__MODULE__` directly. For `audit_log_entries`, I'd have to rename the key in `Map.update`, which means temporarily saving the key, deleting the other key, and then putting in the new key, which I'm not a huge fan of. Should I go with that approach or stick with this one and be more graceful about missing keys?